packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts

```diff
@@ -0,0 +1,217 @@
+			const action = extendedActions[i];
+			if (action.type !== 'action') continue;
+			const label = action.properties?.codex?.label;
+			if (!firstIndexMap.has(label)) {
+				firstIndexMap.set(label, i);
+			}
+		}
+
+		// Keep track of the number of inserted labels
+		let insertedLabels = 0;
+
+		// Create and insert new label objects at the first index of each label
+		for (const label of labelsSet) {
+			const newLabel: LabelCreateElement = {
+				uuid: label,
+				type: 'label',
+				key: label,
+				subcategory: extendedActions[0].key,
+				properties: {
+					key: label,
+				},
+			};
+
+			const insertIndex = firstIndexMap.get(label)! + insertedLabels;
+			extendedActions.splice(insertIndex, 0, newLabel);
+			insertedLabels++;
+		}
+
+		return extendedActions;
+	}
+
+	function parseCategoryActions(
+		actions: INodeCreateElement[],
+		category: string,
+		withLabels = true,
+	) {
+		const filteredActions = filterActionsCategory(actions, category);
+		if (withLabels) return injectActionsLabels(filteredActions);
+		return filteredActions;
+	}
+
+	function getActionData(actionItem: ActionTypeDescription): IUpdateInformation {
+		const displayOptions = actionItem.displayOptions;
+
+		const displayConditions = Object.keys(displayOptions?.show || {}).reduce(
+			(acc: IDataObject, showCondition: string) => {
+				acc[showCondition] = displayOptions?.show?.[showCondition]?.[0];
+				return acc;
+			},
+			{},
+		);
+
+		return {
+			name: actionItem.displayName,
+			key: actionItem.name as string,
+			value: { ...actionItem.values, ...displayConditions } as INodeParameters,
+		};
+	}
+
+	function getNodeTypesWithManualTrigger(nodeType?: string): string[] {
+		if (!nodeType) return [];
+
+		const { selectedView, openSource } = useNodeCreatorStore();
+		const { workflowTriggerNodes } = useWorkflowsStore();
+		const isTrigger = useNodeTypesStore().isTriggerNode(nodeType);
+		const workflowContainsTrigger = workflowTriggerNodes.length > 0;
+		const isTriggerPanel = selectedView === TRIGGER_NODE_CREATOR_VIEW;
+		const isStickyNode = nodeType === STICKY_NODE_TYPE;
+		const singleNodeOpenSources = [
+			NODE_CREATOR_OPEN_SOURCES.PLUS_ENDPOINT,
+			NODE_CREATOR_OPEN_SOURCES.NODE_CONNECTION_ACTION,
+			NODE_CREATOR_OPEN_SOURCES.NODE_CONNECTION_DROP,
+		];
+
+		// If the node creator was opened from the plus endpoint, node connection action, or node connection drop
+		// then we do not want to append the manual trigger
+		const isSingleNodeOpenSource = singleNodeOpenSources.includes(openSource);
+		const shouldAppendManualTrigger =
+			!isSingleNodeOpenSource &&
+			!isTrigger &&
+			!workflowContainsTrigger &&
+			isTriggerPanel &&
+			!isStickyNode;
+
+		const nodeTypes = shouldAppendManualTrigger ? [MANUAL_TRIGGER_NODE_TYPE, nodeType] : [nodeType];
+
+		return nodeTypes;
+	}
+
+	function setAddedNodeActionParameters(
+		action: IUpdateInformation,
+		telemetry: Telemetry,
+		track = true,
+		rootView?: string,
```

Comment:
could we simplify and not pass `telemetry` if not tracking?
```suggestion
		action: IUpdateInformation,
		telemetry?: Telemetry,
		rootView = '',
```

Reply:
Good idea, updated